### ISSUE TITLE
Custom argument types for track events do not get recorded

### ISSUE THREAD

**Issue by hiddenbitious** [google/perfetto#469]
This is more of a question than filing a bug.
I am following the instruction at https://perfetto.dev/docs/instrumentation/track-events but I can't get custom arguments for track events to work.

I have added some simple custom track event arguments of a string and a vector of uint64. You can see the changes here: https://pastebin.com/FC4qUUNX

I am initializing perfetto with
```
#define NDEBUG
#include "perfetto/tracing.h"
#include "protos/perfetto/trace/track_event/gfxr_events.pbzero.h"

PERFETTO_DEFINE_CATEGORIES(perfetto::Category("GFXR").SetDescription("GFXR Queue submissions"));
PERFETTO_TRACK_EVENT_STATIC_STORAGE();
```

and

```
perfetto::TracingInitArgs args;
args.backends |= perfetto::kInProcessBackend;
args.backends |= perfetto::kSystemBackend;
perfetto::Tracing::Initialize(args);
perfetto::TrackEvent::Register();
```

I attempt to fill in the new fields with the following code:
```
TRACE_EVENT_INSTANT("GFXR", "QueueSubmit???", [&](perfetto::EventContext ctx) {
            auto submission = ctx.event()->set_queue_submission();
            submission->set_name(submit_name);
            submission->add_cmd_buf_handles(0x1);
            submission->add_cmd_buf_handles(0xdeadbeaf);
            submission->add_cmd_buf_handles(0xbadc0fee);
            printf("%u\n", __LINE__);
        });
```
The trace is captured. There are event with `QueueSubmit???` name but the other parameters are not captured.
What am I missing?

perfetto revision: d0e70e94

**Comment by ddiproietto** [comment]
> The trace is captured. There are event with QueueSubmit??? name but the other parameters are not captured.
What am I missing?

What do you mean they're not **captured**? They simply do not show up in the UI? That is expected, the rest of the perfetto code does not know anything about your custom new custom protos.

You have two options:

1. Teach trace_processor how to parse your custom protos and put them in the SQL tables. Possibly, change the UI to display the new fields (if any?) in the tables somehow.
2. Avoid using custom protos. Perfetto trace_processor and UI already know how to display "debug annotations". Examples on how to use debug annotation from the SDK here: https://cs.android.com/android/platform/superproject/+/master:external/perfetto/src/tracing/test/api_integrationtest.cc;l=3223;drc=d73666015058abcfad04b0c2208657ba93f51609

To inspect what's in a trace file, you can use the tool `traceconv text` (but you have to rebuild it to make it understand your new protos).

**Comment by hiddenbitious** [comment]
> What do you mean they're not captured? They simply do not show up in the UI? That is expected,

Ah I see. I was expecting that they would just appear in the UI. It was very confusing. Thank you for the clarification
>Avoid using custom protos.

I want to push an arbitrary number of pointers along with the event. How do you suggest achieving this?

**Comment by ddiproietto** [comment]
> I want to push an arbitrary number of pointers along with the event. How do you suggest achieving this?

TL;DR I would just use "debug annotations".

Long answer: it's really a trade-off. You could think about using custom protos, here is a quick list of pros and cons of using custom protos:
  `-` Some core to write and maintain in the .proto schema itself.
  `-` A lot more code to write and maintain in trace_processor
  `-` (Maybe) A lot more code to write and maintain in the UI
  `+` It's "more strongly typed".
  `+` It can be slightly more efficient to serialize (but I don't believe this makes a difference in practice).
